monai/data/dataset.py

```diff
@@ -50,13 +55,120 @@ def __getitem__(self, index):
         return data
 
 
+class PersistentDataset(Dataset):
+    """
```

Comment:
As this PersistentDataset actually provide a complicated and powerful feature, let's update and add more description in doc-string for: `persistent cache theory`, `how to use & example` and `notes`.
And also add your new dataset to: https://github.com/Project-MONAI/MONAI/blob/master/docs/source/data.rst
Thanks.